examples/platforms/simulation/radio.c

```diff
@@ -746,7 +746,18 @@ void radioGenerateAck(void)
         sReceiveFrame.mInfo.mRxInfo.mAckedWithFramePending = true;
     }
 
-    otMacFrameGenerateImmAck(&sReceiveFrame, sReceiveFrame.mInfo.mRxInfo.mAckedWithFramePending, &sAckFrame);
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+    // Use enh-ack for 802154-2015 secured frame
+    if (otMacFrameIsVersion2015(&sReceiveFrame) && otMacFrameIsSecurityEnabled(&sReceiveFrame))
```

Comment:
```suggestion
    if (otMacFrameIsVersion2015(&sReceiveFrame))
```
For 2015 frame, ACK should be enh-ACK. For security, it will be checked in `TxFrame::GenerateEnhAck`